happi/tests/test_client.py

```diff
@@ -351,3 +351,36 @@ def test_client_mapping(
     assert list(dict(client)) == list(three_valves.keys())
     for name in client:
         assert client[name]['name']
+
+
+def test_cache_context(monkeypatch, client_with_three_valves: Client):
+    client = client_with_three_valves
+    valve1 = client['VALVE1']
+    valve2 = client['VALVE2']
+    valve3 = client['VALVE3']
+
+    cache_clear_count = 0
+    orig_clear_cache = client.backend.clear_cache
+
+    def clear_cache():
+        nonlocal cache_clear_count
+        cache_clear_count += 1
+        return orig_clear_cache()
+
+    monkeypatch.setattr(client.backend, "clear_cache", clear_cache)
+    with client.retain_cache_context():
+        results1 = set(client.search_regex(name='valve1|valve2'))
+        results2 = set(client.search_regex(name='valve.*'))
+
+    assert cache_clear_count == 1
+
+    assert valve1 != 'VALVE1'
+    assert valve1 != valve1.metadata
+
+    assert len(results1 & results2) == 2
+    assert valve1 in (results1 & results2)
```

Comment:
These test conditions are from the `hashable_search_results` case right?  Should they go in this test?  If I saw `test_cache_context` fail I wouldn't expect it to be due to set operations.

Reply:
Yes, it's a copy/paste of that test. My thought was "let's test that the cache loads only once and performs identically to the original test".